projects/centre_console/src/can_tx_retry_wrapper.c

```diff
@@ -0,0 +1,56 @@
+#include "can_tx_retry_wrapper.h"
+#include "can_transmit.h"
+#include "log.h"
+
+// forward declaring
+static void prv_try_tx(CanTxRetryWrapperStorage *storage);
+
+StatusCode can_tx_retry_wrapper_init(CanTxRetryWrapperStorage *storage,
+                                     CanTxRetryWrapperSettings *settings) {
+  storage->retries = settings->retries;
+  storage->retry_count = 0;
+  return STATUS_CODE_OK;
+}
+
+static StatusCode prv_can_simple_ack(CanMessageId msg_id, uint16_t device, CanAckStatus status,
+                                     uint16_t num_remaining, void *context) {
+  CanTxRetryWrapperStorage *storage = (CanTxRetryWrapperStorage *)context;
+  if (num_remaining || status) {
+    storage->retry_count++;
+    LOG_DEBUG("retry_count: %d\n", storage->retry_count);
+    if (storage->retry_count >= storage->retries) {
```

Comment:
multiple fault events for a single after retry count and on indefinite retries?

Reply:
well on infinite retries I wanna keep raising faults until it passes... we may have to change this behaviour based on the use case later...

Reply:
But that was the use case I intended when writing this.